lib/quda_matrix.h

```diff
@@ -1005,22 +1028,22 @@ namespace quda {
       w_p = sqrt(c1)*sin(theta*inv3);
       
       //[29] Construct objects for fj = hj/(9u^2 - w^2).
-      double u_sq = u_p*u_p;
-      double w_sq = w_p*w_p;
-      double denom = 9*u_sq - w_sq;
-      double exp_iu_re = cos(u_p);
-      double exp_iu_im = sin(u_p);
-      double exp_2iu_re = exp_iu_re*exp_iu_re - exp_iu_im*exp_iu_im;
-      double exp_2iu_im = 2*exp_iu_re*exp_iu_im;
-      double cos_w = cos(w_p);
-      double sinc_w;
-      double hj_re = 0.0;
-      double hj_im = 0.0;
+      undMatType u_sq = u_p*u_p;
+      undMatType w_sq = w_p*w_p;
+      undMatType denom_inv = 1.0/(9*u_sq - w_sq);
+      undMatType exp_iu_re = cos(u_p);
+      undMatType exp_iu_im = sin(u_p);
+      undMatType exp_2iu_re = exp_iu_re*exp_iu_re - exp_iu_im*exp_iu_im;
+      undMatType exp_2iu_im = 2*exp_iu_re*exp_iu_im;
+      undMatType cos_w = cos(w_p);
+      undMatType sinc_w;
+      undMatType hj_re = 0.0;
+      undMatType hj_im = 0.0;
   
       //[33] Added one more term to the series given in the paper.
       if (w_p < 0.05 && w_p > -0.05) {      
 	//1 - 1/6 x^2 (1 - 1/20 x^2 (1 - 1/42 x^2(1 - 1/72*x^2)))
-	sinc_w = 1.0-1.0/6.0*w_sq*(1-0.05*w_sq*(1-1.0/42.0*w_sq*(1-1.0/72.0*w_sq)));
+	sinc_w = 1.0 - 1.0/6.0*w_sq*(1 - 0.05*w_sq*(1 - (1.0/42.0)*w_sq*(1 - (1.0/72.0)*w_sq)));
```

Comment:
While the order of evaluation may be well defined in C++, it would make it easier to visually parse if brackets were used to clearly delineate the division by 6.0 and subsequent multiplication by `w_sq`.